various improvements for characters

ThomasBreuer opened the pull request:
- adjusted the print functions for character tables to the rules from "Details on printing in Oscar" (and adjusted the manual examples and tests)

- document the output of the text/plain method for `show` (addresses issue #2482)

- fixed the text/latex output in the case that the irrationalities are shown via a legend (and that there are several lines in the legend)

- added more documentation and doctests

- changed `character_field` for Brauer characters: The field is now a *finite field* in this case.

- use the `GAPWrap` approach for more GAP functions

ThomasBreuer committed:
various improvements for characters

- adjusted the print functions for character tables to the rules
  from "Details on printing in Oscar"
  (and adjusted the manual examples and tests)

- document the output of the text/plain method for `show`
  (addresses issue #2482)

- fixed the text/latex output in the case that the irrationalities
  are shown via a legend (and that there are several lines in the legend)

- added more documentation and doctests

- changed `character_field` for Brauer characters:
  The field is now a *finite field* in this case.

- use the `GAPWrap` approach for more GAP functions

ThomasBreuer committed:
move functions, do not copy them ...

fingolfin reviewed (commented):
Test fail, see https://github.com/oscar-system/Oscar.jl/pull/2519/files#file-test-groups-matrixdisplay-jl-L200

ThomasBreuer committed:
further improvements

- fixed a test for `labelled_matrix_formatted`
- renamed `group_class_function` to `class_function`
- added `natural_character` for matrix groups in positive characteristic
  and for group homomorphisms
- admit `IntegerUnion` for the `characteristic` of character tables,
  not just `Int`

ThomasBreuer committed:
fixed doctests

lgoettgens commented:
This needs a rebase.

ThomasBreuer committed:
adjustments/fixes

- adjusted an `is_prime_power_with_data` call
  (The ordering of return values of `is_prime_power_with_data`
  has been changed in Nemocas/Nemo.jl/pull/1484.)

- adjusted the type of a (perhaps empty) list
  (This seemed to work at least up to Julia 1.8.5.)

- fixed some doctests
